File: erlang/cpu_intensive.erl
-module(cpu_intensive).
-compile(export_all).

fib_test() ->
    fib(40),
    fib(40),
    fib(40).

fib(0) -> 1;
fib(1) -> 1;
fib(N) -> fib(N-1) + fib(N-2).

File: lang/erlang/cpu_intensive.c
unsigned int fib(unsigned int n) {
    if (n == 0 || n == 1) {
        return 1;
    }
    return fib(n-1) + fib(n-2);
}

int main() {
    fib(40);
    fib(40);
    fib(40);
    return 0;
}
